File: Laboratorio02/CodigoPython/universidad.py
from curso import Curso


class Universidad:
    def __init__(self, nombre: str):
        self.nombre = nombre
        self.cursos = []

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, valor):
        self._nombre = valor

    @property
    def cursos(self):
        return self._cursos

    @cursos.setter
    def cursos(self, valor):
        self._cursos = valor

    def agregar_curso(self, curso: Curso):
        self.cursos.append(curso)

    def __str__(self):
        return f"Universidad: {self.nombre} , cursos: {self.cursos}"

File: Laboratorio02/CodigoC++/Universidad.cpp
#include "Universidad.h"

#include <string>
#include <vector>
using namespace std;

Universidad::Universidad(string nombre) : nombre(nombre) {}

string Universidad::getNombre() const {
    return nombre;
}

void Universidad::setNombre(string nombre) {
    this->nombre = nombre;
}

vector<Curso> Universidad::getCursos() const {
    return cursos;
}

void Universidad::setCursos(vector<Curso> cursos) {
    this->cursos = cursos;
}

void Universidad::agregarCurso(Curso curso) {
    cursos.push_back(curso);
}

string Universidad::toString() const {
    string resultado = "Universidad: " + nombre + " , cursos: [";
    for (size_t i = 0; i < cursos.size(); ++i) {
        if (i > 0) {
            resultado += ", ";
        }
        resultado += cursos[i].toString();
    }
    resultado += "]";
    return resultado;
}
